src/ResultSet.php

```diff
@@ -3,45 +3,106 @@
 namespace ipl\Orm;
 
 use ArrayIterator;
+use BadMethodCallException;
+use Generator;
 use Iterator;
 use Traversable;
 
+/**
+ * Dataset containing database rows
+ */
 class ResultSet implements Iterator
 {
+    /** @var ArrayIterator */
     protected $cache;
 
     /** @var bool Whether cache is disabled */
     protected $isCacheDisabled = false;
 
+    /** @var Generator */
     protected $generator;
 
+    /** @var ?int */
     protected $limit;
 
+    /** @var ?int */
     protected $position;
 
-    public function __construct(Traversable $traversable, $limit = null)
+    /** @var ?int */
+    protected $offset;
+
+    /** @var ?int */
+    protected $pageSize;
+
+    /**
+     * Construct the {@see self::class class} object
```

Comment:
PHPStorm doesn't quite understand this:
![image](https://github.com/Icinga/ipl-orm/assets/16668527/daf293cb-0262-4a0a-bf84-2f72cfc27cdd)

I think that a reference is unnecessary anyway. It's the docstring of the class anyway. And only of this class, not that of a parent. So, just use the plain name.

Reply:
It does work by clicking on the class link, but you're right. An overloaded class would use a custom docstring anyways. Will be removed.